### PR TITLE
Update octoprint plugins

### PR DESCRIPTION
**Pull request by tri-ler**
## Description of changes

- add: maintainers: add tri-ler

- octoprint.plugins.printtimegenius: 2.3.1 -> 2.3.3

- octoprint.python.pkgs.firmwareupdater: init at 1.14.0
- octoprint.python.pkgs.fullscreen: init at 0.0.6
- octoprint.python.pkgs.mqttchambertemperature: init at 0.0.3
- octoprint.python.pkgs.navbartemp: init at 0.15
- octoprint.python.pkgs.obico: init at 2.5.0
- octoprint.python.pkgs.octopod: init at 0.3.16
- octoprint.python.pkgs.prusaslicerthumbnails: init at 1.0.7
- octoprint.python.pkgs.resource-monitor: init at 0.3.16
- octoprint.python.pkgs.timelapsepurger: init at 0.1.4

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by arcnmx** [comment]
:+1: with python now updated to 3.12 this is needed for the plugins to run properly

**Review by arcnmx — APPROVED** [review]
:+1: with python now at 3.12 this is needed for the updated plugin to run properly

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 1]
```diff
@@ -231,15 +271,100 @@ in
...
+
+    propagatedBuildInputs = with super; [
+      backoff
+      sentry-sdk
+      bson
+      distro
+    ];
+
+    meta = with lib; {
+      description = "Securely monitor and control your OctoPrint-connected printer from anywhere for free with Obico. Get unlimited live webcam streaming, full OctoPrint remote access, printer status notifications, and a free companion mobile app for iOS and Android. The best part? AI-powered failure detection watches your prints so you don’t have to.";
```
Comment (gador) [inline pkgs/applications/misc/octoprint/plugins.nix:331]:
```suggestion
      Description = "Monitor Octoprint-connected printers with Obico";
      longDescription = "Securely monitor and control your OctoPrint-connected printer from anywhere for free with Obico. Get unlimited live webcam streaming, full OctoPrint remote access, printer status notifications, and a free companion mobile app for iOS and Android. The best part? AI-powered failure detection watches your prints so you don’t have to.";
```

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 2]
```diff
@@ -261,6 +386,27 @@ in
...
+      owner = "jneilliii";
+      repo = "OctoPrint-PrusaSlicerThumbnails";
+      rev = version;
+      sha256 = "sha256-waNCTjAZwdBfhHyJCG2La7KTnJ8MDVuX1JLetFB5bS4=";
+    };
+
+    propagatedBuildInputs = with super; [ psutil  ];
+
+    meta = with lib; {
+      description = "Plugin that extracts thumbnails from uploaded gcode files sliced by PrusaSlicer.";
```
Comment (gador) [inline pkgs/applications/misc/octoprint/plugins.nix:403]:
```suggestion
      description = "Plugin that extracts thumbnails from uploaded gcode files sliced by PrusaSlicer";
```

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 3]
```diff
@@ -289,6 +435,27 @@ in
...
+      owner = "Renaud11232";
+      repo = "OctoPrint-Resource-Monitor";
+      rev = version;
+      sha256 = "sha256-w1PBxO+Qf7cSSNocu7BiulZE7kesSa+LGV3uJlmd0ao=";
+    };
+
+    propagatedBuildInputs = with super; [ psutil  ];
+
+    meta = with lib; {
+      description = "A plugin to view the current CPU and RAM usage on your system";
```
Comment (gador) [inline pkgs/applications/misc/octoprint/plugins.nix:452]:
```suggestion
      description = "Plugin to view the current CPU and RAM usage on your system";
```

File: pkgs/applications/misc/octoprint/plugins.nix [hunk 4]
```diff
@@ -367,6 +534,25 @@ in
...
+
+    src = fetchFromGitHub {
+      owner = "jneilliii";
+      repo = "OctoPrint-TimelapsePurger";
+      rev = version;
+      sha256 = "sha256-XS4m4KByScGTPfVE4kuRLw829gNE2CdM0RyhRqGGxyw=";
+    };
+
+    meta = with lib; {
+      description = "Automatically deletes timelapses that are older than configured timeframe.";
```
Comment (gador) [inline pkgs/applications/misc/octoprint/plugins.nix:549]:
```suggestion
      description = "Automatically deletes timelapses that are older than configured timeframe";
```

**Review by gador — CHANGES_REQUESTED** [review]
Looks good to me. Just a few nitpicks

File: pkgs/applications/misc/octoprint/plugins.nix (hunk 4 above)
Comment (tri-ler) [reply pkgs/applications/misc/octoprint/plugins.nix:549]: Would like to keep it consistent with the other plugins, as well as keeping it the same as the official plugin's repo

File: pkgs/applications/misc/octoprint/plugins.nix (hunk 3 above)
Comment (tri-ler) [reply pkgs/applications/misc/octoprint/plugins.nix:452]: Would like to keep it consistent with the other plugins, as well as keeping it the same as the official plugin's repo

File: pkgs/applications/misc/octoprint/plugins.nix (hunk 1 above)
Comment (tri-ler) [reply pkgs/applications/misc/octoprint/plugins.nix:331]: Would like to keep it consistent with the other plugins

**Comment by tri-ler** [comment]
@gador 

Hey, thanks for taking a look. I think it makes more sense to me to keep the formatting the same across the plugins as well as keeping it the same as the plugin's official repo.

**Comment by gador** [comment]
> Hey, thanks for taking a look. I think it makes more sense to me to keep the formatting the same across the plugins as well as keeping it the same as the plugin's official repo.

I completely understand your reasoning and motivation! However there still should be some changes to keep thinks consistent across `nixpkgs`:

- No period at the end of a description
- _Short_ description. Please just one sentence
- Not start with an article

It is fine if you reword upstreams description. It should reflect what the plugin does, but this not have to be a word-by-word copy of the description upstream. Also I'm fine if you do not want the keep the `longDescription` in `obico`

For reference, there are attempts to normalize these things across `nixpkgs` (eg #317959) and also these recommendations are mentioned here https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes so I'm inclined to "do it right" the first time.

**Comment by tri-ler** [comment]
> > Hey, thanks for taking a look. I think it makes more sense to me to keep the formatting the same across the plugins as well as keeping it the same as the plugin's official repo.
> 
> I completely understand your reasoning and motivation! However there still should be some changes to keep thinks consistent across `nixpkgs`:
> 
> * No period at the end of a description
> * _Short_ description. Please just one sentence
> * Not start with an article
> 
> It is fine if you reword upstreams description. It should reflect what the plugin does, but this not have to be a word-by-word copy of the description upstream. Also I'm fine if you do not want the keep the `longDescription` in `obico`
> 
> For reference, there are attempts to normalize these things across `nixpkgs` (eg #317959) and also these recommendations are mentioned here https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes so I'm inclined to "do it right" the first time.

Hey, 

Thanks for showing me this, I did not know. I definetly agree about the need to normalize things and doing it right the first time. I have updated the PR with the changes you suggested.

**Comment by gador** [comment]
I just tested all your plugins locally in a vm with octoprint:

- `mqttchambertemperature` fails to load due to `No module named jsonpath_ng`

The rest build and run as expected :+1: 
Maybe you have another look at `mqttchambertemperature` ?

**Comment by tri-ler** [comment]
I was not able to replicate it, but it looks like the plugin creator added it to the plugin requirements now. I updated this to relfelc it, could you check if it works for you now?

**Review by gador — APPROVED** [review]
Works now :+1: 
Looks good to me!

**Pull request merged**